fineract-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/service/InterestRefundService.java

```diff
@@ -0,0 +1,35 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.fineract.portfolio.loanaccount.service;
+
+import java.math.BigDecimal;
+import java.time.LocalDate;
+import org.apache.fineract.portfolio.loanaccount.domain.Loan;
+
+public interface InterestRefundService {
+
+    boolean canHandle(Loan loan);
+
+    BigDecimal calculateTotalPaidInterest(Loan loan);
+
+    BigDecimal calculateInterestRefundAmount(Loan loan, BigDecimal relatedRefundTransactionAmount, LocalDate relatedRefundTransactionDate,
+            BigDecimal totalPaidInterest, BigDecimal totalPayableInterestBeforeTransaction);
```

Comment:
Frankly speaking the existing codebase works with tons of methods with a lot of parameters. Since the types are the same, I'd say a parameter object here would make sense - especially that it can be easily extended in the future. Thoughts?

Also, consider it for the other methods as well just for consistency.